addons/website/models/mixins.py

```diff
@@ -144,9 +146,35 @@ def open_website_url(self):
             'target': 'self',
         }
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super(WebsitePublishedMixin, self).create(vals_list)
+
+        modifies_website_published = any('website_published' in values for values in vals_list)
+        if modifies_website_published and not all(record.can_publish for record in res):
+            raise AccessError(_("You do not have the rights to publish/unpublish"))
+
+        return res
+
+    @api.multi
+    def write(self, values):
+        if 'website_published' in values and not all(record.can_publish for record in self):
+            raise AccessError(_("You do not have the rights to publish/unpublish"))
```

Comment:
What about having a method returning the string so that people can customize the displayed message ? For example in slides it would be something like "Publish is restricted to the responsible of training courses or members of publisher group for documentation course"

Reply:
Nice idea, it's done.